src/shared/components/comment/comment-report.tsx

```diff
@@ -161,6 +176,49 @@ export class CommentReport extends Component<
             />
           )}
         </button>
+        <ActionButton
+          label={I18NextService.i18n.t(
+            comment_view.comment.removed ? "restore_comment" : "remove_comment",
+          )}
+          inline
+          icon={comment_view.comment.removed ? "restore" : "x"}
+          noLoading
+          onClick={() => this.setState({ showRemoveCommentDialog: true })}
```

Comment:
I really don't like not having the handlers be separate functions, like we've done everywhere else in this codebase, but I'm not gonna block because of it.